### PR TITLE
Build on GHC 8.8

### PR DESCRIPTION
**Pull request by Avi-D-coder**
This builds, but segfaults if I try to interact with a displayed graph.

ezgif-6-6e4c9172bdac

### CODE REVIEW

File: stack.yaml [hunk 1]
```diff
@@ -1,6 +1,18 @@
-resolver: lts-8.8
+resolver: nightly-2020-01-06
 packages:
 - '.'
 extra-deps:
-- xdot-0.3.0.1
-- svgcairo-0.13.2.0
+  - git: https://github.com/def-/xdot.git
```
Comment (def-) [inline stack.yaml]: why not 0.3.0.2?

Comment (Avi-D-coder) [reply stack.yaml]: The latest hackage release is `0.3.0.1`, `xdot` needs to be updated to support 8.8 before this PR lands.
I had switched to the git version in the hope that it had fixes for the segfaults.

**Comment by Avi-D-coder** [comment]
The issue now is `free(): double free detected in tcache 2` when interacting with the graph mode. Thoughts?

**Comment by Avi-D-coder** [comment]
I suspect that all the `MVar` are racy

**Comment by Avi-D-coder** [comment]
Still WIP
It's still crashing there are a number places that could still hold races, but I think it be a different issue

**Comment by Avi-D-coder** [comment]
The stm/less IO everywhere branch is [stm](https://github.com/Avi-D-coder/ghc-vis/commits/stm) if your interested, but it did not fix the error.

Removing `widgetQueueDraw canvas` causes makes `view 1` not error, but obviously it also prevents rendering. I'm not familiar with GTK so I don't know if this is a GTK or heap view issue.

**Comment by nomeata** [comment]
I can confirm that it builds and crashes, already on mouse-over

**Comment by nomeata** [comment]
Yes, with https://github.com/rcook/xdot/commit/b373370a39287613e728e3abc4d875905bb17ed4 applied to xdot I am making progress

**Comment by def-** [comment]
Thanks for looking into this. I'll merge fixes if you have them.

**Comment by nomeata** [comment]
With https://github.com/def-/xdot/pull/11 merged into `xdot` and that released to hackage, this PR should be good in a good shape to go in.

**Comment by def-** [comment]
Done, thanks.

**Pull request merged**